### ISSUE TITLE
[Help request]build dragonfly version 1.0 api 

### ISSUE THREAD

**Issue by yunfeiyanggzq** [dragonflyoss/Dragonfly#1082]
## Why you need it?

Now supernode add dfget use 0.3 version api to download file. But the v0.3 api  can't work well for many situations. So our community want to update the api.
## How it could be?

The community has done many thing for the  v0.1 api.The v1.0 api run as the [picture](https://github.com/dragonflyoss/Dragonfly/blob/master/docs/images/design/dfget_supernode.png) shows.And the community has done many work about the version1.0 api.such as [some supernode v1.0 bridge](https://github.com/dragonflyoss/Dragonfly/blob/9017881ccebde40b1ea47b875b71c4122b13333e/supernode/server/router.go#L54),
now we sould to finish the following work to make v1.0 api working.
1. finish the supernode v1.0 bridge.

peer:
- POST /peers register a peer [done]
- DElETE /peer/{id} [done]
- GET /peer/{id} [done]
- GET /peers [done]

task:
- POST: /tasks
- DELETE: /tasks/{id} [done]
- GET:/task/{id}
- GET:/task

piece: 
- GET:/tasks/{id}/piece
- GET:/tasks/{id}/pieces/{peiceRange}
- GET:/tasks/{id}/pieces/{pieceRange}/error [done] 

About the api's function detail you can see this Dragonfly client file.
*NOTE:*The done bridges haven't been test,so there may be something wrong with it.

2.finish the dfget's api 
- /peers register a peer
- /tasks register a task
- /tasks/{id}/pieces get schedule result 
- /tasks/{id}/pieces/{peiceRange} update piece result
- /peer/{id} /shut peer down

we should change the dfget's logic.because dfget  runs in a different way.

**Comment by jyizheng** [comment]
I am happy to look at the task API of the supernode bridge.

**Comment by Starnop** [comment]
ref: #1065

**Comment by yunfeiyanggzq** [comment]
> I am happy to look at the task API of the supernode bridge.
emmm,I am sorry, I have modify some api of supernode in https://github.com/dragonflyoss/Dragonfly/issues/1082#issuecomment-554751325.we can work together to finish the dfget.the work is 
Tiǎozhàn de
3/5000
Challenging

**Comment by yunfeiyanggzq** [comment]
of course,there are many problems with the supernode bridge.you can modify it

**Comment by zhaxzhax** [comment]
keep focusing on it

**Comment by zhaxzhax** [comment]
In order to realize a dynamic bandwidth limit [issue#1427](https://github.com/dragonflyoss/Dragonfly/issues/1427), I wanna add a dfget api `/peer/{id}/peerState/{dynamicRate}` to allow dfget server to send the dynamic totalLimit to supernode which can be used to the newest schedule algorithm (sort peers by `dynamicRate/ProducerLoad`).
So I'm not sure that if it's ok to add a new dfget api `/peer/{id}/peerState/{dynamicRate}` in 1.0 version.
If you think it's required also, it's thankful that you could update the new api in this issue.
